### PR TITLE
Node systems fixes + improvements

### PR DESCRIPTION
**Pull request by salman-codewar**
- [x] Moved `notify` service to `node-system`
- [x] Cleanup `notification-system`
- [x] `node-system` linting and build fixes
- [x] `node-system` services communicate with other system via `init-client`
- [x] Updated `node-system` services to use common providers
- [x] Workflow updated
- [x] Update deploy script to deploy node system locally

### CODE REVIEW

File: systems/node/node-gateway/pkg/rest/router.go [hunk 1]
```diff
@@ -55,17 +56,26 @@ type RouterConfig struct {
 }
 
 type Clients struct {
-	Health health
+	health Health
```
Comment (Brackleycassinga) [inline systems/node/node-gateway/pkg/rest/router.go:59]: Clients should be as follow :
```
type Clients struct {
    Health  health
    Notify  notify
}
```

File: systems/node/node-gateway/pkg/rest/router.go [hunk 2]
```diff
@@ -55,17 +56,26 @@ type RouterConfig struct {
...
+
+type Health interface {
 	StoreRunningAppsInfo(req *healthPb.StoreRunningAppsInfoRequest) (*healthPb.StoreRunningAppsInfoResponse, error)
 	GetRunningAppsInfo(nodeId string) (*healthPb.GetRunningAppsResponse, error)
 }
 
 func NewClientsSet(endpoints *pkg.GrpcEndpoints) *Clients {
 	c := &Clients{}
-	c.Health = client.NewHealth(endpoints.Health, endpoints.Timeout)
+	c.health = client.NewHealth(endpoints.Health, endpoints.Timeout)
```
Comment (Brackleycassinga) [inline systems/node/node-gateway/pkg/rest/router.go:78]: This should be :
```
c := &Clients{}
	c.Health= client.NewHealth(endpoints.Network, endpoints.Timeout)
	c.Notify= client.NewNode(endpoints.Node, endpoints.Timeout)
```

File: systems/node/node-gateway/pkg/rest/router.go [hunk 3]
```diff
@@ -195,7 +212,7 @@ func (r *Router) postSystemPerformanceInfoHandler(c *gin.Context, req *StoreRunn
 		genCapps = append(genCapps, genCapp)
 	}
 
-	return r.clients.Health.StoreRunningAppsInfo(&healthPb.StoreRunningAppsInfoRequest{
+	return r.clients.health.StoreRunningAppsInfo(&healthPb.StoreRunningAppsInfoRequest{
```
Comment (Brackleycassinga) [inline systems/node/node-gateway/pkg/rest/router.go:215]: This should be :
`r.clients.Health.StoreRunningAppsInfo(`

File: systems/node/node-gateway/pkg/rest/router.go [hunk 4]
```diff
@@ -204,7 +221,7 @@ func (r *Router) postSystemPerformanceInfoHandler(c *gin.Context, req *StoreRunn
 }
 
 func (r *Router) getSystemPerformanceInfoHandler(c *gin.Context, req *GetRunningAppsRequest) (*healthPb.GetRunningAppsResponse, error) {
-	resp, err := r.clients.Health.GetRunningAppsInfo(req.NodeId)
+	resp, err := r.clients.health.GetRunningAppsInfo(req.NodeId)
```
Comment (Brackleycassinga) [inline systems/node/node-gateway/pkg/rest/router.go:224]: It should be 
`r.clients.Health`

File: systems/registry/site/pkg/config.go [hunk 5]
```diff
@@ -35,7 +35,7 @@ type Config struct {
 }
 
 type HttpServices struct {
-	InventoryClient string `defaut:"localhost:8080"`
+	InventoryClient string `defaut:"http://api-gateway-inventory:8080"`
```
Comment (Brackleycassinga) [inline systems/registry/site/pkg/config.go:38]: Did you test this with `HttpServices` change ?

**Review by Brackleycassinga — CHANGES_REQUESTED** [review]
Thanks for adding the common client to Node sys. I added a few comments for review.

File: systems/node/configurator/cmd/server/main.go [hunk 6]
```diff
@@ -80,16 +82,24 @@ func runGrpcServer(gormdb sql.Db) {
 		instanceId = inst.String()
 	}
 
-	reg := providers.NewRegistryProvider(serviceConfig.RegistryHost, serviceConfig.DebugMode)
+	regUrl, err := ic.GetHostUrl(ic.CreateHostString(serviceConfig.OrgName, "registry"), serviceConfig.Http.InitClient, &serviceConfig.OrgName, serviceConfig.DebugMode)
+	if err != nil {
+		log.Errorf("Failed to resolve registry address: %v", err)
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/node/configurator/cmd/server/main.go:87]: Here we're checking error for `init-client/getHostUrl`.

File: systems/node/controller/pkg/server/controller.go [hunk 7]
```diff
@@ -68,22 +72,26 @@ func (c *ControllerServer) RestartSite(ctx context.Context, req *pb.RestartSiteR
...
 
-	if err := c.registrySystem.ValidateSite(netId.String(), req.GetSiteName(), c.orgName); err != nil {
+	_, err = c.siteClient.Get(req.GetSiteName())
+
+	if err != nil {
 		return nil, fmt.Errorf("failed to validate site %s and network %s. Error %s", req.GetSiteName(), netId.String(), err.Error())
 	}
 
-	if err := c.registrySystem.ValidateNetwork(netId.String(), c.orgName); err != nil {
+	_, err = c.networkClient.Get(netId.String())
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/node/controller/pkg/server/controller.go:81]: `Get` is a get network by id req. How can we get network without passing id?
You can check the provider implementation under `common/rest/registry/network`

File: systems/registry/site/pkg/config.go (hunk 5 above)
Comment (salman-codewar) [reply systems/registry/site/pkg/config.go:38]: yes

File: systems/node/node-gateway/pkg/rest/router.go (hunk 4 above)
Comment (salman-codewar) [reply systems/node/node-gateway/pkg/rest/router.go]: addressed

File: systems/node/node-gateway/pkg/rest/router.go (hunk 3 above)
Comment (salman-codewar) [reply systems/node/node-gateway/pkg/rest/router.go]: addressed

File: systems/node/node-gateway/pkg/rest/router.go (hunk 2 above)
Comment (salman-codewar) [reply systems/node/node-gateway/pkg/rest/router.go]: addressed

File: systems/node/node-gateway/pkg/rest/router.go (hunk 1 above)
Comment (salman-codewar) [reply systems/node/node-gateway/pkg/rest/router.go]: addressed

File: systems/node/controller/pkg/server/controller.go [hunk 8]
```diff
@@ -68,22 +72,26 @@ func (c *ControllerServer) RestartSite(ctx context.Context, req *pb.RestartSiteR
...
+
+	if err != nil {
 		return nil, fmt.Errorf("failed to validate network with network %s. Error %s", netId.String(), err.Error())
 	}
 
-	nodes, err := c.registrySystem.GetNodesBySite(req.SiteName)
+	nodes, err := c.nodeClient.GetNodesBySite(req.SiteName)
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/node/controller/pkg/server/controller.go]: addressed

File: systems/node/controller/pkg/server/controller.go [hunk 9]
```diff
@@ -68,22 +72,26 @@ func (c *ControllerServer) RestartSite(ctx context.Context, req *pb.RestartSiteR
 		return nil, status.Errorf(codes.InvalidArgument, "invalid network ID format: %s", err.Error())
 	}
 
-	if err := c.registrySystem.ValidateSite(netId.String(), req.GetSiteName(), c.orgName); err != nil {
+	_, err = c.siteClient.Get(req.GetSiteName())
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/node/controller/pkg/server/controller.go]: addressed

File: systems/node/controller/pkg/server/controller.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (Brackleycassinga) [reply systems/node/controller/pkg/server/controller.go:81]: We don't need network validation as network will not be part of request anymore.

**Review by Brackleycassinga — APPROVED** [review]
LGTM 👍🏾

**Pull request merged**